Sources/Commands/SwiftPackageTool.swift

```diff
@@ -727,3 +743,34 @@ private extension Diagnostic.Message {
         .error("missing required argument \(argument)")
     }
 }
+
+fileprivate extension SwiftPackageTool {
+    /// Logs all changed dependencies to a stream
+    /// - Parameter changes: Changes to log
+    /// - Parameter pins: PinsStore with currently pinned packages to compare changed packages to.
+    /// - Parameter stream: Stream used for logging
+    func logPackageChanges(changes: [(PackageReference, Workspace.PackageStateChange)], pins: PinsStore, on stream: OutputByteStream = stdoutStream) {
+        let changes = changes.filter { $0.1 != .unchanged }
+        
+        stream <<< "\n"
+        stream <<< "\(changes.count) dependenc\(changes.count == 1 ? "y has" : "ies have") changed\(changes.count > 0 ? ":" : ".")"
+        stream <<< "\n"
+        
+        for (package, change) in changes {
+            guard let packageName = package.name else { continue }
+            let currentVersion = pins.pinsMap[package.identity]?.state.description ?? ""
+            switch change {
+            case let .added(requirement):
+                stream <<< "+ \(packageName) \(requirement.prettyPrinted)"
+            case let .updated(requirement):
+                stream <<< "~ \(packageName) \(currentVersion) -> \(packageName) \(requirement.prettyPrinted)"
+            case .removed:
+                stream <<< "- \(packageName) \(currentVersion)"
+            default:
+                continue
+            }
+            stream <<< "\n"
+        }
+        stream.flush()
```

Comment:
Can you post what an example output looks like? Just curious!

Reply:
3 dependencies have changed:
+ SwiftProtobuf 1.7.0
~ BigInt 4.0.0 -> BigInt 5.0.0
- swift-nio 2.8.0